analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ParameterValidationInAsyncShouldBeWrapped.CSharp10.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Tests.Diagnostics
+{
+    public class InvalidCases
+    {
+        public static async Task<string> FooAsync(string something) // Noncompliant
+        {
+            ArgumentNullException.ThrowIfNull(something);
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Secondary
+
+            await Task.Delay(1);
+            return something + "foo";
+        }
+
+        public static async Task<string> ThrowExpressionAsync(string something) // FN
```

Comment:
I think it makes sense to document the associated issue #6369
```suggestion
        public static async Task<string> ThrowExpressionAsync(string something) // FN https://github.com/SonarSource/sonar-dotnet/issues/6369
```